include/swift/AST/DiagnosticsSema.def

```diff
@@ -1601,6 +1601,11 @@ ERROR(inferred_opaque_type,none,
 // Extensions
 ERROR(non_nominal_extension,none,
       "non-nominal type %0 cannot be extended", (Type))
+WARNING(multiple_nominals_in_extended_type,none,
+        "expected a single nominal type in the extended type position but"
+        " found several. This may lead to unexpected errors while"
```

Comment:
Nit: Could you move the space to the line above and remove the period at the end? (look at other multi-line diagnostics for example). Also, perhaps the wording of the warning could be simplified?

```
WARNING(multiple_nominals_in_extended_type,none,
        "extending several types at once via protocol composition " 
        "may lead to unexpected errors while type-checking. Use a "
        "'where' clause for additional constraints", ())

Reply:
I fully agree with the sentiment that the current diagnostic is too technical/complex. However, "extending several types at once" is not quite accurate as the overall composition might simplify to a single type.

Reply:
How about

> extending protocol compositions and typealiases of protocol compositions may lead to unexpected errors while type-checking. Instead, pick a single type to be extended and use a 'where' clause for additional constraints.

Is that better?

Reply:
Yeah, that’s looks much better to me. Perhaps the last bit could be “checking; choose a single type...”.

Reply:
I think "this may lead to unexpected errors while type-checking" sounds a bit ominous. I would prefer if the diagnostic just said "extending a protocol composition is not supported; extending '%0' instead" where %0 is the first nominal.